src/test/FuzzerImpl.cpp

```diff
@@ -1510,6 +1519,10 @@ TransactionFuzzer::reduceTrustLineBalancesAfterSetup(
         // Reduce "account"'s balance of this asset by paying the
         // issuer.
         auto tle = stellar::loadTrustLine(ltx, account, asset);
+        if (!tle.isAuthorizedToMaintainLiabilities())
```

Comment:
`isAuthorizedToMaintainLiabilities` only returns false if don't have either `AUTHORIZED_FLAG` or `AUTHORIZED_TO_MAINTAIN_LIABILITIES_FLAG` set, so the balance adjustment would fail. This is required because `getAvailableBalance` will throw if the trustline is not authorized.

Reply:
You're missing the point: the balance adjustment has to happen, or the contract is broken. So how can you make that work?

Reply:
Ah I see what you're saying. In this case, the trustline balance should be 0, and the available balance expected should also be 0, so I added a check for both. 

Related to this, the fuzzer should be updated at some point to allow for an allow trust call at the end so we can test trustline liabilities and balances while in different authorization states.